weave/tests/trace/test_weave_client.py

```diff
@@ -1446,3 +1447,49 @@ def test_object_version_read(client):
     )
     assert obj_res.obj.val == {"a": 5}
     assert obj_res.obj.version_index == 5
+
```

Comment:
I would add a clause to this test that loops through all the known digests and validates that they return the correct payload